python/arcticdb/storage_fixtures/s3.py

```diff
@@ -0,0 +1,275 @@
+"""
+Copyright 2023 Man Group Operations Limited
+
+Use of this software is governed by the Business Source License 1.1 included in the file LICENSE.txt.
+
+As of the Change Date specified in that file, in accordance with the Business Source License, use of this software will be governed by the Apache License, version 2.0.
+"""
+
+import logging
+import multiprocessing
+import json
+import os
+import re
+import requests
+from typing import NamedTuple, Optional, Any
+
+from .api import *
+from .utils import get_ephemeral_port, GracefulProcessUtils, wait_for_server_to_come_up
+from arcticc.pb2.storage_pb2 import EnvironmentConfigsMap
+from arcticdb.version_store.helper import add_s3_library_to_env
+
+# All storage client libraries to be imported on-demand to speed up start-up of ad-hoc test runs
+
+Key = NamedTuple("Key", [("id", str), ("secret", str), ("user_name", str)])
+
+logging.getLogger("botocore").setLevel(logging.INFO)
+
+
+class S3Bucket(StorageFixture):
+    _FIELD_REGEX = {
+        ArcticUriFields.HOST: re.compile("^s3://()([^:/]+)"),
+        ArcticUriFields.BUCKET: re.compile("^s3://[^:]+(:)([^?]+)"),
+        ArcticUriFields.USER: re.compile("[?&](access=)([^&]+)(&?)"),
+        ArcticUriFields.PASSWORD: re.compile("[?&](secret=)([^&]+)(&?)"),
+    }
+
+    key: Key
+    _boto_bucket: Any = None
+
+    def __init__(self, factory: "BaseS3StorageFixtureFactory", bucket: str):
+        super().__init__()
+        self.factory = factory
+        self.bucket = bucket
+
+        if factory.enforcing_permissions:
+            self.key = factory._create_user_get_key(bucket + "_user")
```

Comment:
`BaseS3StorageFixtureFactory` doesn't have `_create_user_get_key` so this confuses my linter. Slightly neater to just have a polymorphic `get_key` method on the factory? Similar issue for `factory._iam_admin` and `factory._RW_POLICY` and `factory._RO_POLICY` below

Reply:
_If_ we one-day support permissions on `BaseS3StorageFixtureFactory`, then it will have to implement these as well. I can add a type check for now.